rs/index/src/utils.rs

```diff
@@ -33,3 +34,33 @@ impl SearchContext {
         self.visited_pages.as_ref().unwrap().len()
     }
 }
+
+#[derive(Debug)]
+pub struct IdWithScore {
+    pub id: u64,
+    pub score: f32,
+}
+
+impl Ord for IdWithScore {
+    fn cmp(&self, other: &Self) -> Ordering {
+        // We want to use this in a min-heap, so we reverse the order.
+        self.score
+            .partial_cmp(&other.score)
+            .unwrap_or(Ordering::Equal)
+            .reverse()
+    }
+}
```

Comment:
You should have the reverse to be elsewhere, otherwise if someone else uses it, it will be very confusing.

I think you can do `Reverse(IdWithScore{})` or something

Reply:
Yeah good point